internal/cmd/gocdk/add_test.go

```diff
@@ -0,0 +1,195 @@
+// Copyright 2019 The Go Cloud Development Kit Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package main
+
+import (
+	"context"
+	"io/ioutil"
+	"log"
+	"net"
+	"net/http"
+	"net/url"
+	"os"
+	"path/filepath"
+	"strings"
+	"testing"
+)
+
+const mainContent = `
+package main
+
+import (
+  "log"
+	"net/http"
+  "os"
+)
+
+func main() {
+	port := os.Getenv("PORT")
+  log.Fatal(http.ListenAndServe(":" + port, nil))
+}
+`
+
+func TestPortableTypeDemos(t *testing.T) {
+	dir, cleanup, err := newTestModule()
+	if err != nil {
+		t.Fatal(err)
+	}
+	defer cleanup()
+
+	pctx := &processContext{
+		workdir: dir,
+		env:     os.Environ(),
+		stdout:  ioutil.Discard,
+		stderr:  ioutil.Discard,
+	}
+	// Uncomment when debugging.
+	// pctx.stdout = os.Stdout
+	// pctx.stderr = os.Stderr
+	ctx := context.Background()
+
+	// TODO(rvangent): Run gocdk init instead of this.
+	if err := ioutil.WriteFile(filepath.Join(dir, "main.go"), []byte(mainContent), 0666); err != nil {
```

Comment:
`newTestModule` should do what you want.

Reply:
I call `newTestModule` up above. But it currently doesn't include a `main.go` file, which is what this line is adding.